back/engines/commercial/idea_custom_fields/app/controllers/idea_custom_fields/web_api/v1/admin/idea_custom_fields_controller.rb

```diff
@@ -113,21 +114,23 @@ def update_fields
             field = CustomField.new update_params.merge(resource: @custom_form)
             SideFxCustomFieldService.new.before_create(field, current_user)
             if field.save
-              temp_ids_to_ids_mapping[field_params[:temp_id]] = field.id if field_params[:temp_id]
+              page_temp_ids_to_ids_mapping[field_params[:temp_id]] = field.id if field_params[:temp_id]
             else
               errors[index.to_s] = field.errors.details
               next
             end
             SideFxCustomFieldService.new.after_create(field, current_user)
           end
-          update_options field, options_params, errors, index if options_params
+          if options_params
+            option_temp_ids_to_ids_mapping = update_options field, options_params, errors, index
```

Comment:
What happens when we have two fields with options to update? It looks to me that the `option_temp_ids_to_ids_mapping` for the first field to update will be overwritten by the `option_temp_ids_to_ids_mapping` for the second field to update.

Reply:
Very good catch! Thanks. Elaborated the test to send 2 new options with logic on their value, and fixed the bug.